src/webui/api/rsscontroller.cpp

```diff
@@ -91,6 +94,29 @@ void RSSController::itemsAction()
     setResult(jsonVal.toObject());
 }
 
+void RSSController::markAsReadAction()
+{
+    requireParams({"itemPath"});
+
+    const QString itemPath {params()["itemPath"]};
+    const QString articleId {params()["articleId"]};
+
+    RSS::Item *item = RSS::Session::instance()->itemByPath(itemPath);
+    if (!item) return;
+
+    if (params().contains("articleId")) {
```

Comment:
If corresponding parameter is missing you have "null-string" in articleId. So you can use `if (!articleId.isNull())` here to avoid double lookup.